velox/common/memory/Memory.h

```diff
@@ -97,6 +105,29 @@ struct MemoryManagerOptions {
   /// same query on all the workers instead of a random victim query which
   /// happens to trigger the failed memory arbitration.
   bool retryArbitrationFailure{true};
+
+  /// A factory function to create memory arbitrator. This defines how the
+  /// arbitrator in memory manager is created and gives possibility to user
+  /// to use custom implementation of memory arbitrator.
+  ///
+  /// By default the factory directly calls MemoryArbitrator::create.
+  ///
+  /// If customized implementation is used, MemoryManagerOptions.arbitratorKind
+  /// should also be set to Kind::kCustom too to let Velox recognize the
+  /// arbitrator as a non-builtin implementation.
+  MemoryArbitratorFactory arbitratorFactory{
```

Comment:
```
/// The customized arbitrator create method which is only set in case of custom kind of arbitrator.
MemoryArbitratorFactory arbitratorFactory{nullptr}
```